### ISSUE TITLE
Improve installation instructions (conda env?)

### ISSUE THREAD

**Issue by kbarnhart** [landlab/landlab#632]
@mcflugen @gregtucker @SiccarPoint @nicgaspar @margauxmouchene @nathanlyons @erkanistan @saisiddu 

We've had a recent slew of installations issues with conda that seem to have the theme of dependency conflicts. We've also had a few users who've started to use conda environments but seem to struggle to remember that these environments must be activated (and have packages re-installed/associated with them). 

I wanted to start a discussion about how we should address this. 

One approach that might be worth trying is to change the landlab.github.io installation instructions to say something like. 

***
Like many python packages, Landlab has some dependencies. While the development team has worked to minimize these dependencies in order to make Landlab as compatible as possible with other python packages, there are some packages that we must rely on that are often the source of compatibility issues (e.g NetCDF reading packages). 

If you get a dependency conflict when installing Landlab we recommend the following:

1) Install Landlab in a conda environment 

[Insert instructions here (OSX and Windows), include activate/deactivate, links to conda env documentation, and reminder to also install jupyer notebooks, etc in the environment.]

2) Post an issue on GitHub so the development team can keep track of the dependency conflicts our users are experiencing. 
***

thoughts?

**Comment by mcflugen** [comment]
@kbarnhart I actually think our installation instructions are pretty good as-is. They're short and to the point.

However, maybe we should have a link to an FAQ or a troubleshooting guide in case folks run into problems. We definitely have some questions that we receive frequently enough to warrant one. This could also just be instructing users to search the issues before creating a new issue.

Something else that may help is to hide the instructions about how to test landlab. Since we already test our releases, this is mostly just for developers and users don't really need to do this.  More often than not (always?) when users see tests that fail, they're false positives (for instance, the tests for the current releases will fail with numpy 1.14 but landlab works just fine).

**Comment by margauxmouchene** [comment]
I agree that the installation instructions should stay very short on the landlab.github.io page.

While installing in a conda environment is a great solution to many install problems on messy computers, I think it should not be the default install instructions.

I have started making a list of all the installation problems (from github issues, reports on Slack and our conversations): I am going to make a 'troubleshooting' wiki page with this list and links to the solutions. Conda environment will be one of them. And, at the bottom of the page, if they don't find a solution to their problem in this list, how to post an issue.

**Comment by mcflugen** [comment]
Thanks @margauxmouchene! That sounds great. As you are going through the issues, when you find an issue that is about an installation question or problem, could you please label it as "install" (it's a new label I just created). We can then point users to [all issues that are labeled as "install"](https://github.com/landlab/landlab/issues?utf8=%E2%9C%93&q=is%3Aissue+label%3Ainstall). If we remember to keep using that label, perhaps it will make it easier for folks to find a solution.

**Comment by nicgaspar** [comment]
@kbarnhart @mcflugen @margauxmouchene 

Katy - we have gone back and forth on how much install info to give on the splash page. We have settled on minimal. We don't have to stay at minimal, but that is the history. So I guess you might receive pushback to add this. I do like Eric's suggesting of linking to an FAQ or Troubleshooting page.

Eric - I am fully on board for removing the direction to run landlab tests. Maybe we just tell them to do an import, and if it fails, then go to troubleshooting. The tests are confusing because a lot of information and warnings come with them. So I wonder if they are just generally confusing people.

Margaux - thanks for organizing this info.

**Comment by kbarnhart** [comment]
I'm definitely a fan of minimal (but discoverable). My concern is that issues exist on Github, but users don't know to search for them. I love @margauxmouchene Troubleshooting wiki and @mcflugen's tags! 

What about this approach?

Add one sentence to the *Installing Landlab Using Anaconda* section of the main website. 

"If you experience issues installing Landlab using Anaconda refer to the Troubleshooting guide on the Landlab Wiki or look at existing Github issues related to installation." (with two links to the two resources). 

That way there is some direction from the splash page regarding what to do if the conda install fails.

**Comment by mcflugen** [comment]
@kbarnhart: How about just have a link to the Troubleshooting Guide? From there we can provide instructions and links to other resources.

*"Problems installing landlab? Please see our [Troubleshooting Guide](https://github.com/landlab/landlab/wiki/Troubleshooting)."*

**Comment by kbarnhart** [comment]
that sounds good to me.

**Comment by margauxmouchene** [comment]
[Troubleshooting Guide](https://github.com/landlab/landlab/wiki/Troubleshooting) is done, listing the problems users have reported, the solutions we offered and links to related Github issues.

I also added 'install' labels to all relevant Github issues.

Most problems are due to:

- dependencies: conda environments seem to be the best solution in this case, any other suggestions?

- path configuration: I discovered that there is an awesome [page](https://github.com/landlab/landlab/wiki/Correcting-Install-Paths) on the wiki just for that! I added links to this page in the Troubleshooting guide.

- recently, tests failing with `AttributeError: 'LandlabTester' object has no attribute 'check_fpu_mode'`, due to numpy version 1.14 (see #616). I agree with @kbarnhart and @mcflugen that tests are probably confusing for most users, I removed the instructions to test when [installing with Anaconda](https://github.com/landlab/landlab/wiki/Installing-Landlab-with-Anaconda) (non-developer install). If they get an error when importing and running Landlab, users should refer now to the Troubleshooting guide.

**Comment by kbarnhart** [comment]
@margauxmouchene nice work!
I'm happy to close this issue unless others think it should stay open. 

THoughts?
